### ISSUE TITLE
A puzzle about the regression targets of anchors which have no intersection with any ground-truth boxes

### ISSUE THREAD

**Issue by 2696120622** [fizyr/keras-retinanet#1236]
According the logic for computing the overlap in compute_overlap.pyx, if an anchor has no intersections with any ground-truth boxes, then, its overlaps with each ground-truth boxes will be zero and be assigned a background with state 0. However, in function bbox_transform, its regression targets are computed as the offset between it and one ground-truth box as the np.argmax(overlaps, axis=1) return a index of one ground-truth box. As a result, it has a state 0 of  background, but its box regression targets is the offset between it and one ground-truth box. It is odd. Do you know why?

**Comment by hgaiser** [comment]
You are correct, however we ignore the background bboxes in the loss function (not the cleanest way to do it, but anything else becomes quite complicated): https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/losses.py#L96

**Comment by 2696120622** [comment]
Thanks! I get it.
